### PR TITLE
Implement Security Service

### PR DESCRIPTION
**Pull request by SyntacticStructures**
Enables access to the following Token APIs:
* Create Token
* Get Tokens
* Refresh Token
* Revoke Token

### CODE REVIEW

File: tests/artifactorysecurity_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	params.RefreshToken = token.RefreshToken
+	params.AccessToken = token.AccessToken
+	newToken, err := testsSecurityService.RefreshToken(params)
+	if err != nil {
+		t.Error(err)
+	}
+	tokensToRevoke = append(tokensToRevoke, newToken.RefreshToken)
+}
+
+func getTokensTest(t *testing.T) {
```
Comment (SyntacticStructures) [inline tests/artifactorysecurity_test.go]: This test ~doubles the runtime of the test suite in my experience, due to the performance of the artifactory endpoint. I'm not sure if it's going to be a welcome addition or if it should be removed.

**Comment by eyalbe4** [comment]
Thanks so much for this @SyntacticStructures!
Can you please join our contributors community by signing [JFrog's CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) so that we can merge your code?

**Comment by SyntacticStructures** [comment]
Thank you very very much for taking a look @eyalbe4.

> Can you please join our contributors community by signing JFrog's CLA so that we can merge your code?

I have signed the CLA.

**Comment by eyalbe4** [comment]
@SyntacticStructures,
I like the code and the tests!
When running the test though, I got the following error:
```
=== RUN   TestToken
=== RUN   TestToken/CreateToken
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token
=== RUN   TestToken/RevokeToken
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token/revoke
=== RUN   TestToken/RevokeToken:_token_not_found
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token/revoke
=== RUN   TestToken/RefreshToken
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token
=== RUN   TestToken/GetTokens
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token
[Debug] Sending HTTP GET request to: http://localhost:8081/artifactory/api/security/token
[Info] REVOKING ALL  3  tokens
[Debug] Revoking Token:  9637c15d-9b79-42dd-b171-7c17a49e67f2
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token/revoke
[Debug] Revoking Token:  
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token/revoke
[Error] Artifactory response: 400 Bad Request
{
  "error": "invalid_request",
  "error_description": "token or token_id are required"
}
[Error] Token was not revoked:  
[Debug] Revoking Token:  87a9b4cf-02f2-4fd5-b8d4-b3b58447070b
[Debug] Sending HTTP POST request to: http://localhost:8081/artifactory/api/security/token/revoke
--- FAIL: TestToken (0.07s)
    --- PASS: TestToken/CreateToken (0.01s)
    --- PASS: TestToken/RevokeToken (0.02s)
    --- PASS: TestToken/RevokeToken:_token_not_found (0.00s)
    --- FAIL: TestToken/RefreshToken (0.01s)
        artifactorysecurity_test.go:69: Artifactory response: 400 Bad Request
            {
              "error": "invalid_request",
              "error_description": "Provided access token with ID '10a226c0-dc18-4ca2-a0cc-73ee7aaba117' was issued by a different service with ID 'jfrt@01cw3k21r2hesy14c3kdg505qt' (current service ID: 'jfrt@01ddstt02echb50cnhm5k512w1')"
            }
    --- PASS: TestToken/GetTokens (0.01s)
```

**Comment by SyntacticStructures** [comment]
@eyalbe4 Thanks for posting details about the failed test. I will try to reproduce.

As for the CLA, I signed it a couple of weeks ago when I first opened this PR. I don't remember details about how I filled it out. I will just sign it again and pay better attention this time.

**Comment by SyntacticStructures** [comment]
Alright @eyalbe4, the issue should be fixed now: ed1b311bb.

Looks like the issue arose from the fact that I am testing against an Artifactory with only one instance.

**Comment by eyalbe4** [comment]
Thanks @SyntacticStructures.
Did you sign the [CLA](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N)?

**Comment by SyntacticStructures** [comment]
Yes @eyalbe4 I signed the CLA.

**Comment by eyalbe4** [comment]
Thanks @SyntacticStructures - 
With which name did you sign it?

**Comment by SyntacticStructures** [comment]
I signed it as "Taylor Lea".

**Pull request merged**